modules/fedimint-wallet/src/lib.rs

```diff
@@ -699,29 +733,31 @@ impl ServerModule for Wallet {
         dbtx: &mut ModuleDatabaseTransaction<'_, ModuleInstanceId>,
         output: &WalletOutput,
     ) -> Result<TransactionItemAmount, ModuleError> {
-        if !output
-            .recipient
-            .is_valid_for_network(self.cfg.consensus.network)
-        {
-            return Err(WalletError::WrongNetwork(
-                self.cfg.consensus.network,
-                output.recipient.network,
-            ))
-            .into_module_error_other();
-        }
-        let consensus_fee_rate = self.current_round_consensus(dbtx).await.unwrap().fee_rate;
-        if output.fees.fee_rate < consensus_fee_rate {
-            return Err(WalletError::PegOutFeeRate(
-                output.fees.fee_rate,
-                consensus_fee_rate,
-            ))
-            .into_module_error_other();
+        if let WalletOutput::PegOut(pegout) = output {
+            if !pegout
+                .recipient
+                .is_valid_for_network(self.cfg.consensus.network)
+            {
+                return Err(WalletError::WrongNetwork(
+                    self.cfg.consensus.network,
+                    pegout.recipient.network,
+                ))
+                .into_module_error_other();
+            }
+            let consensus_fee_rate = self.current_round_consensus(dbtx).await.unwrap().fee_rate;
+            if pegout.fees.fee_rate < consensus_fee_rate {
+                return Err(WalletError::PegOutFeeRate(
+                    pegout.fees.fee_rate,
+                    consensus_fee_rate,
+                ))
+                .into_module_error_other();
+            }
```

Comment:
Why don't we validate at least that the RBFed tx exists in case of `WalletOutput::Rbf`?

Reply:
That will fail during `create_peg_out_tx` a few lines down.